src/main/java/org/folio/inventory/support/JsonHelper.java

```diff
@@ -53,4 +70,37 @@ private String readFile(InputStream is) throws IOException {
       return sb.toString();
     }
   }
+
+  private static JsonObject handleNullNestedFields(JsonObject itemObject) {
+    var keysToRemove = new ArrayList<String>();
+    for (var key : itemObject.fieldNames()) {
+      var value = itemObject.getValue(key);
+      if (value == null) {
+        keysToRemove.add(key);
+      } else if (value instanceof String && StringUtil.isEmpty((String) value)) {
+        keysToRemove.add(key);
+      } else if (value instanceof JsonObject) {
+        handleNullNestedFields((JsonObject) value);
+      }
+    }
+
+    keysToRemove.forEach(itemObject::remove);
+    return itemObject;
+  }
+
+  private static List<Iterable<?>> toNotNullList(Collection<?> collection) {
+    return collection.stream()
+      .filter(item -> item != null)
```

Comment:
You can use as an alternative lambda expression. For example,
filter(Objects::nonNull)